runtime/dom-particle.js

```diff
@@ -120,6 +121,26 @@ class DomParticle extends XenStateMixin(Particle) {
       slot.render({});
     }
   }
+  renderHostedSlot(slotName, hostedSlotId, content) {
+    assert(this.hostedSlotBySlotId.has(hostedSlotId), `Missing model for slot ID ${hostedSlotId}`);
+    this.hostedSlotBySlotId.get(hostedSlotId).content = content;
+
+    let slot = this.getSlot(slotName);
+    if (slot) {
+      let combinedContent = this.combineHostedContent(Object.keys(content));
+
+      // TODO: group multiple calls together!
+      if (combinedContent) {
+        slot.render(combinedContent);
+      } else if (slot.isRendered) {
+        slot.render({});
+      }
+    }
+  }
+  combineHostedContent(contentTypes) {
+    // Method must be implemented by transformation particle.
+    // TODO: consider adding a default implementation.
```

Comment:
I think that'd be a good idea, yeah. Maybe basically the implementation above.

Reply:
I was thinking that it should be in a transformation particle base class, which is not necessarily dom-particle at all.
Now that I look at it, I realize that this applies to renderHostedSlot above as well.
So probably both - renderHostedSlot and combineHostedContent should be abstract methods of Particle, and implemented by transformation particles only.
WDYT?

Reply:
I think the lowest level API should be directly responding to events on slots. This is a nice abstraction over that, which is why I thought it made sense to put it in DOMParticle. However, your suggestion of a TransformationParticle superclass might be a better bet - or maybe a mixin even.

Reply:
so the multiplexing particle still depends on some of the dom-particle functionality, so for now i just moved the combineHostedContent implementation to dom-particle. And added a TODO to reconsidering having a transformation-particle base class later.